junit-runtime/src/main/scala/org/scalajs/junit/Reporter.scala

```diff
@@ -58,16 +58,23 @@ private[junit] final class Reporter(eventHandler: EventHandler,
     }
 
     if (errors.nonEmpty) {
-      emit(errors.head)
+      val errorsPatchedForAssumptionViolations = errors.map {
+        case error: internal.AssumptionViolatedException =>
+          new TestCouldNotBeSkippedException(error)
+        case error =>
+          error
+      }
```

Comment:
I don't think this is the right place to do this. IMO `executeTests` / `runTests` in `JUnitTask` is the right place for this.

Reply:
I wasn't sure. I moved it to `JUnitTask`.